data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/resync/CascadingActionDetector.java

```diff
@@ -0,0 +1,187 @@
+                            } else if (foreignKeyRelation.getUpdateAction() == ForeignKeyAction.SET_DEFAULT) {
+                                createResyncPartition(
+                                        foreignKeyRelation.getDatabaseName(),
+                                        foreignKeyRelation.getChildTableName(),
+                                        foreignKeyRelation.getForeignKeyName(),
+                                        foreignKeyRelation.getForeignKeyDefaultValue(),
+                                        tableMetadata.getPrimaryKeys(),
+                                        event.getHeader().getTimestamp());
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Detects if a binlog event contains cascading deletes and if detected, creates resync partitions
+     */
+    public void detectCascadingDeletes(Event event, Map<String, ParentTable> parentTableMap, TableMetadata tableMetadata) {
+        if (parentTableMap.containsKey(tableMetadata.getFullTableName())) {
+            final ParentTable parentTable = parentTableMap.get(tableMetadata.getFullTableName());
+
+            for (String column : parentTable.getColumnsWithCascadingDelete().keySet()) {
+                for (ForeignKeyRelation foreignKeyRelation : parentTable.getColumnsWithCascadingDelete().get(column)) {
+                    if (foreignKeyRelation.getDeleteAction() == ForeignKeyAction.CASCADE) {
+                        LOG.warn("Cascade delete is not supported yet");
+                    } else if (foreignKeyRelation.getDeleteAction() == ForeignKeyAction.SET_NULL) {
+                        // foreign key in the child table will be set to NULL
+                        createResyncPartition(
+                                foreignKeyRelation.getDatabaseName(),
+                                foreignKeyRelation.getChildTableName(),
+                                foreignKeyRelation.getForeignKeyName(),
+                                null,
+                                tableMetadata.getPrimaryKeys(),
+                                event.getHeader().getTimestamp());
+                    } else if (foreignKeyRelation.getDeleteAction() == ForeignKeyAction.SET_DEFAULT) {
+                        createResyncPartition(
+                                foreignKeyRelation.getDatabaseName(),
+                                foreignKeyRelation.getChildTableName(),
+                                foreignKeyRelation.getForeignKeyName(),
+                                foreignKeyRelation.getForeignKeyDefaultValue(),
+                                tableMetadata.getPrimaryKeys(),
+                                event.getHeader().getTimestamp());
+                    }
+                }
+            }
+        }
+    }
+
+    private String getFullTableName(String database, String table) {
+        return database + "." + table;
```

Comment:
We use the delimiter "." in multiple places. Move this to constant to make it reusable.